### PR TITLE
Minor fixes for cesm port

### PR DESCRIPTION
**Pull request by jedwards4b**
scripts_regression_test
Test suite: 
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes

User interface changes?: 

Code review: jgfouca

### CODE REVIEW

File: utils/python/CIME/case.py [hunk 1]
```diff
@@ -277,6 +277,8 @@ def set_value(self, item, value, subgroup=None, ignore_type=False):
         then that value will be set in the file object and the file
         name is returned
         """
+        if item == "CASEROOT":
```
Comment (jedwards4b) [inline utils/python/CIME/case.py]: We now have a _caseroot variable in the case object - make sure that it is updated if set_value("CASEROOT" is called.

File: utils/python/CIME/system_test.py [hunk 2]
```diff
@@ -243,8 +243,15 @@ def _log_output(self, test, output):
     ###########################################################################
     def _get_case_id(self, test):
     ###########################################################################
-        baseline_action_code = ".C" if self._compare else (".G" if self._generate else "")
-        return "%s%s.%s" % (test, baseline_action_code, self._test_id)
+        baseline_action_code = ""
```
Comment (jedwards4b) [inline utils/python/CIME/system_test.py]: cesm allows both compare and generate flags, this should be reflected in testname

File: utils/python/CIME/system_test.py [hunk 3]
```diff
@@ -835,7 +842,11 @@ def _setup_cs_files(self):
                     fd.write(template)
                 os.chmod(cs_submit_file,
                          os.stat(cs_submit_file).st_mode | stat.S_IXUSR | stat.S_IXGRP)
-
+            if CIME.utils.get_model == "cesm":
```
Comment (jedwards4b) [inline utils/python/CIME/system_test.py]: testreporter is a cesm tool to talk to the test database

File: utils/python/wait_for_tests.py [hunk 4]
```diff
@@ -11,6 +11,7 @@
 TEST_PENDING_STATUS       = "PEND"
 TEST_PASS_STATUS          = "PASS"
 TEST_FAIL_STATUS          = "FAIL"
+NO_BASELINE_STATUS    = "BFAIL"
```
Comment (jedwards4b) [inline utils/python/wait_for_tests.py]: BFAIL indicates that the baseline was not found

**Comment by jedwards4b** [comment]
annotation complete

File: utils/python/wait_for_tests.py (hunk 4 above)
Comment (jgfouca) [inline utils/python/wait_for_tests.py]: I may need to do some additional work in wait_for_tests to properly support this test status.

**Comment by jgfouca** [comment]
Looks good.

**Pull request merged**